core/frontend/src/components/vehiclesetup/configuration/failsafes/Failsafes.vue

```diff
@@ -0,0 +1,200 @@
+import autopilot_data from '@/store/autopilot'
+
+import { FailsafeDefinition } from './types'
+
+export default {
+  name: 'FailsafesConfigration',
+  components: {
+    failsafeCard,
+  },
+  data() {
+    return {
+      failsafes: [] as FailsafeDefinition[],
+    }
+  },
+  computed: {
+    params_finished_loaded(): boolean {
+      return autopilot_data.finished_loading
+    },
+  },
+  mounted() {
+    // we fetch the failsafes on mounting because we
+    // need to be in an async context to wait for the import calls
+    this.get_failsafes().then((failsafes) => {
+      this.failsafes = failsafes
+    })
+  },
+  methods: {
+    async get_failsafes(): Promise<FailsafeDefinition[]> {
+      const failsafes: FailsafeDefinition[] = [
+        {
+          name: 'Ground Station Heartbeat Loss',
+          generalDescription: 'The heartbeat failsafe is triggered when the vehicle does not receive '
+          + 'a heartbeat from the GCS within 3 seconds.',
+          image: (await import('@/assets/img/configuration/failsafes/heartbeat.svg')).default as string,
+
+          params: [
+            {
+              replacementTitle: 'Timeout',
+              icon: 'mdi-timer',
+              name: 'FS_GCS_TIMEOUT',
+              optional: true,
+            },
+            {
+              replacementTitle: 'Action',
+              icon: 'mdi-lightning-bolt',
+              name: 'FS_GCS_ENABLE',
+              replacementDescription: '',
+            },
+          ],
+        },
+        {
+          name: 'Pilot Input Loss',
+          generalDescription: 'The pilot input failsafe is triggered when the vehicle does not receive '
+          + 'any pilot input for a given amount of time.',
+          image: (await import('@/assets/img/configuration/failsafes/pilot-input.svg')).default as string,
+          params: [
+            {
+              replacementTitle: 'Action',
+              icon: 'mdi-lightning-bolt',
+              name: 'FS_PILOT_INPUT',
+            },
+            {
+              replacementTitle: 'Timeout',
+              icon: 'mdi-timer-outline',
+              name: 'FS_PILOT_TIMEOUT',
+            },
+          ],
+        },
+        {
+          name: 'Leak detection',
+          generalDescription: 'Triggered when a leak is detected. We recoomend keeping control electronics dry.',
+          image: (await import('@/assets/img/configuration/failsafes/leak.svg')).default as string,
+          params: [
+            {
+              replacementTitle: 'Action',
+              icon: 'mdi-lightning-bolt',
+              name: 'FS_LEAK_ENABLE',
+            },
+            {
+              replacementTitle: 'Leak probe 1 type',
+              name: 'LEAK1_TYPE',
+            },
+            {
+              replacementTitle: 'Leak 1 pin',
+              name: 'LEAK1_PIN',
+            },
+            {
+              replacementTitle: 'Leak 1 logic level when dry',
+              name: 'LEAK1_LOGIC',
+            },
```

Comment:
Is this temporary? If so, might want to add a code comment about it or something. Per an earlier review comment, I think the Action should typically be the last field, since it's conditional on the other fields.

For the record, we've discussed having the pin management and configuration handled in a separate "Leak Sensor" configuration page, in which case it's less problematic to have the parameters for all three supported connections, while also simplifying what needs to get shown in the failsafe card (which is just the failsafe action, and preferably a link to the sensor configuration page).

Reply:
yeah my goal is to allow links to "advanced" settings wherever applicable, right now I have in mind doing this for battery and leak at least

Reply:
and maybe keeping this simpler version here, not 100% sure about that

Reply:
> and maybe keeping this simpler version here, not 100% sure about that

I think it's cleaner to separate the concerns - the Action is (high level) failsafe configuration, but the pins and logic level stuff are (lower level) hardware configuration, so can be relegated to the page for that.

Reply:
makes sense. but this can wait until we have that page, right? I'll add a TODO